in_toto/verifylib.py

```diff
@@ -240,6 +270,64 @@ def verify_layout_expiration(layout):
     raise LayoutExpiredError("Layout expired")
 
 
+def substitute_parameters(layout, parameter_dictionary):
+  """
+  <Purpose>
+    This function is a transitionary measure for parameter substitution (or
+    any other solution defined by the in-toto team). As of now, it acts as 
+    a very simple replacement layer for python-like parameters
```

Comment:
It might be good to discuss the use cases for this functionality.  Why and when should an adopter use the substitution layer?  

Also, is this functionality usable from the command-line tools?  This pull request appears to only support it from `verifylib.py`.